### DISCUSSION TITLE
Load grid in nested tabs using entityid from master dialog

### DISCUSSION THREAD

**Discussion by rhemasystems** [serenity-is/Serenity#7086] [Help]
Please I have the below Master dialog that has 3 main tabs, one of the tabs has another set of tabs with grids in each tab.
How can i load the grids in the nested tabs using the entityid of the master dialog
In the 1st image below, am able to load the first tab (**Core Church Life**) using the master entityid because its not a grid, but in the 2nd image the next tab (**Groupings & Associations**) is a grid 

![image](https://github.com/serenity-is/Serenity/assets/50826372/c980f8c0-3591-4f64-ade7-c5ed33468824)

![image](https://github.com/serenity-is/Serenity/assets/50826372/d4625721-5b1b-4d5b-8366-015ccb5b1b7a)

I am using the below to load the **Core Church Life** tab

```
constructor() {
    super();
    this.churchlifePropertyGrid = new Serenity.PropertyGrid(this.byId("ChurchlifePropertyGrid"), {
        idPrefix: this.idPrefix + "_Member_",
        items: Q.getForm(Indotalent.Member.MemberChurchlifeForm.formKey),
        useCategories: true
        //.filter(x => x.name != 'ChurchMemberIds')
    });
    this.visitationGrid = new MemberVisitationGrid(this.byId('VisitationGrid'));
}

protected loadEntity(entity: MemberRow) {
    super.loadEntity(entity);
    var thisform = this;
    Serenity.TabsExtensions.setDisabled(this.tabs, 'Churchlife', this.isNewOrDeleted());
    Serenity.TabsExtensions.setDisabled(this.tabs, 'Visitation', this.isNewOrDeleted());


    this.visitationGrid.memberid = entity.Id;
    if (!entity.Id) {
        this.churchlifePropertyGrid.load({});
        return;
    }
    var churchlifeId = this.getChurchLifeID();
    if (!churchlifeId) {
        this.activityid = 0;
        return null;
    }

    if (Q.isEmptyOrNull(churchlifeId.toString()))
        return null;

    MemberChurchlifeService.Retrieve({
        EntityId: churchlifeId
    }, response => {
        this.churchlifePropertyGrid.load(response.Entity);
        this.activityid = Q.toId(response.Entity.Id);
    });
   

}
```

**Comment by rhemasystems** [comment]
@VictorTomaili and @volkanceylan, can you assist me, I think my case is a unique case

**Comment by rhemasystems** [comment]
@VictorTomaili, @volkanceylan , please any help?
Thank you

**Comment by VictorTomaili** [comment]
Hello @rhemasystems 
This is not a unique case. We have many customers with more specific businesses. And all the samples exist to achieve your result. You can check "Grid Filtered By Criteria" sample and adapt it to your dialog. You can pass the filter value to your grid.

**Comment by rhemasystems** [comment]
I am able to pass the filtered value to the grid in the last tab as seen below.
My issue is getting that to work for the grid in the sub tab **Groupings & Associations** of the main tab **Church life**

![image](https://github.com/serenity-is/Serenity/assets/50826372/601191e6-5e67-4040-86d1-c5dc21fd6f1d)

**Reply by LSNicholls** [reply]
Hi,

The sub-tabs shouldn't make any difference, assuming that your Grouping/Associations entity is in the same relation to Member as your Visitations entity -- are you saying that the Grouping/Associations is a grandchild rather than a child of Member?  If so, I have done this -- but the difference has nothing to do with the fact that the grandchild is on a sub-tab.

Then the only difference the sub-tab could *conceivably* make is an accident of timing.  To see if this is the case, try changing the order of the sub-tabs so that Grouping/Associations is the first sub-tab in the set and loads along with the dialog.  Does it work when you do that?  If this test works, then you have to experiment with exactly when that data is supposed to load and whether it has the requisite parent ID available at that time.

I will also caution you -- having done something similar to this -- that you want to make sure you don't accidentally load *all* Grouping/Associations rather than the related ones because the equality filter is missing at some point.  We used the onViewSubmit method of the child Grid -- which btw is in the Grid Filtered By Criteria sample that Victor pointed out -- although there might have been a neater way in our case.

I am not completely following your code, but to be honest I am wondering what happens when you issue this line of code:

`this.visitationGrid.memberid = entity.Id;`

... and I am not seeing the same type of assignment for the grouping/associations grid, shouldn't it be there?  And what happens in the * grid.ts code, when that value is assigned?

**Comment by rhemasystems** [comment]
Hi @LSNicholls , the code below is what loads the visitation records for the selected member into the grid in the **Visitation** tab
`this.visitationGrid.memberid = entity.Id;`

**Reply by LSNicholls** [reply]
Yes, I know what that code does.  But there are two things I can't tell from that code:

1. what is the similar code that does the same thing for the grouping/associations grid?  There should be something similar, right?  Assuming they are both related to Member, right?
2. for both grids, there should be code within the grid.ts files that *reacts* to the changed value of the memberid, right?

**Reply by LSNicholls** [reply]
@rhemasystems , did you solve your problem ?  Otherwise, do you have answers for my questions?
